extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -63,36 +63,17 @@ public static function onAfterInitialize( Title $title, $article, OutputPage $ou
 	}
 
 	/**
-	 * Handle downgrading anonymous requests for our sitemaps.
+	 * Handle downgrading anonymous requests for our sitemaps and robots.txt.
 	 *
-	 * @param  string     $subpage Specific sitemap being requested.
 	 * @param  WebRequest $request
 	 * @param  User       $user
 	 * @return boolean
 	 */
-	public static function onSitemapPageBeforeOutput( string $subpage, WebRequest $request, User $user ): bool {
-		global $wgScriptPath;
+	public static function onSitemapRobotsPageBeforeOutput( WebRequest $request, User $user ): bool {
 		if ( WebRequest::detectProtocol() === 'https' &&
 			!self::httpsAllowed( $user, $request->getFullRequestURL() )
 		) {
-			self::downgradeRedirectForPath( "$wgScriptPath/$subpage", $request );
-			return false;
-		}
-		return true;
-	}
-
-	/**
-	 * Handle downgrading anonymous requests for robots.txt.
-	 *
-	 * @param  WebRequest $request
-	 * @param  User       $user
-	 * @return boolean
-	 */
-	public static function onWikiaRobotsBeforeOutput( WebRequest $request, User $user ): bool {
-		if ( WebRequest::detectProtocol() === 'https' &&
-			!self::httpsAllowed( $user, $request->getFullRequestURL() )
-		) {
-			self::downgradeRedirectForPath( '/robots.txt', $request );
+			self::downgradeRedirectForPath( $request->getFullRequestURL(), $request );
```

Comment:
For the downgrade, do we need to pass both, the path and the request? Maybe we can just pass the `$request` and rename the method to something like `downgradeRedirect` or `downgradeRequest`.

Reply:
No. Currently we don't. 0b8e14a5902